cdap-examples/WikipediaPipeline/src/test/java/co/cask/cdap/examples/wikipedia/WikipediaPipelineAppTest.java

```diff
@@ -0,0 +1,208 @@
+      "successful in the first half of the 1990s, due mainly to the release of [[Nirvana (band)|Nirvana]]'s " +
+      "''[[Nevermind]]'', [[Pearl Jam]]'s ''[[Ten (Pearl Jam album)|Ten]]'', [[Soundgarden]]'s " +
+      "''[[Badmotorfinger]]'', [[Alice in Chains]]' ''[[Dirt (Alice in Chains album)|Dirt]]'', and " +
+      "[[Stone Temple Pilots]]' ''[[Core (Stone Temple Pilots album)|Core]]''.\"}]}}}}";
+    rawWikipediaStreamManager.send(data1);
+    rawWikipediaStreamManager.send(data2);
+  }
+
+  private void testWorkflowWithDefaultThreshold(WorkflowManager workflowManager) throws Exception {
+    workflowManager.start();
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+    String pid = getLatestPid(workflowManager.getHistory());
+    WorkflowTokenNodeDetail tokenAtCondition =
+      workflowManager.getTokenAtNode(pid, WikipediaPipelineWorkflow.EnoughDataToProceed.class.getSimpleName(),
+                                     WorkflowToken.Scope.USER, "result");
+    boolean conditionResult = Boolean.parseBoolean(tokenAtCondition.getTokenDataAtNode().get("result"));
+    Assert.assertFalse(conditionResult);
+    assertWorkflowToken(workflowManager, pid, false);
+  }
+
+  private void testWorkflowWithReducedThreshold(WorkflowManager workflowManager) throws Exception {
+    workflowManager.start(ImmutableMap.of("min.page.threshold", String.valueOf(1)));
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+    String pid = getLatestPid(workflowManager.getHistory());
+    WorkflowTokenNodeDetail tokenAtCondition =
+      workflowManager.getTokenAtNode(pid, WikipediaPipelineWorkflow.EnoughDataToProceed.class.getSimpleName(),
+                                     WorkflowToken.Scope.USER, "result");
+    boolean conditionResult = Boolean.parseBoolean(tokenAtCondition.getTokenDataAtNode().get("result"));
+    Assert.assertTrue(conditionResult);
+    assertWorkflowToken(workflowManager, pid, true);
+  }
+
+  @Nullable
+  private String getLatestPid(List<RunRecord> history) {
+    String pid = null;
+    long latestStartTime = 0;
+    for (RunRecord runRecord : history) {
+      // ok to use start ts, since we ensure that the next run begins after the previous run finishes in the test
+      if (runRecord.getStartTs() > latestStartTime) {
+        latestStartTime = runRecord.getStartTs();
+        pid = runRecord.getPid();
+      }
+    }
+    return pid;
+  }
+
+  private void assertWorkflowToken(WorkflowManager workflowManager, String pid,
+                                   boolean continueConditionSucceeded) throws NotFoundException {
+    assertTokenAtPageTitlesMRNode(workflowManager, pid);
+    assertTokenAtRawDataMRNode(workflowManager, pid, continueConditionSucceeded);
+    assertTokenAtNormalizationMRNode(workflowManager, pid, continueConditionSucceeded);
+    assertTokenAtSparkLDANode(workflowManager, pid, continueConditionSucceeded);
+    // TODO: Verification fails if the last node of the workflow is a fork node.
```

Comment:
I didn't add it here, because #3311 should fix this. I'll test it against that and add a ticket here if it still fails.